request.js

```diff
@@ -357,6 +357,37 @@ Request.prototype.init = function (options) {
     delete self.url
   }
 
+  // If there's a baseUrl, then use it as the base URL (i.e. uri must be
+  // specified as a relative path and is appended to baseUrl).
+  if (self.baseUrl) {
+    if (typeof self.baseUrl !== 'string') {
+      return self.emit('error', new Error('options.baseUrl must be a string'))
+    }
+
+    if (typeof self.uri !== 'string') {
+      return self.emit('error', new Error('options.uri must be a string when using options.baseUrl'))
```

Comment:
we probably should say that "uri must be a string" instead of "options.uri", since we can just pass the uri value directly.